### PR TITLE
improved prime number generator to check only up to sqrt(n) instead of n

### PR DESCRIPTION
**Pull request by steven7214**
### **Describe your change:**
The old algorithm checked to see if a number, n, was divisible by any number 2 - (n-1), but by pigeon hole principle it's only necessary to check up to sqrt(n). I changed the algorithm so it wouldn't do unnecessary checks.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
This is cool!  Can we rename the current function __slow_primes()__ and the new one __primes()__ and then we can land this one.  Then I will create a benchmark (or you can if you want) to demonstrate how much of an improvement this is.  Thanks!

**Comment by cclauss** [comment]
```
./maths/prime_numbers.py:26:34: E226 missing whitespace around arithmetic operator
        bound = int(math.sqrt(i))+1
                                 ^
1     E226 missing whitespace around arithmetic operator
```

**Comment by steven7214** [comment]
Ok, I've added the previous code in a separate function called slow_primes(). You'll notice that the code between the two is very similar; the only difference is that the search has been restricted in the faster method. Is this what you were looking for?

File: maths/prime_numbers.py [hunk 1]
```diff
@@ -2,6 +2,33 @@
...
+    []
+    >>> list(primes(-10))
+    []
+    >>> list(primes(25))
+    [2, 3, 5, 7, 11, 13, 17, 19, 23]
+    >>> list(primes(11))
+    [2, 3, 5, 7, 11]
+    >>> list(primes(33))
+    [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31]
+    >>> list(primes(10000))[-1]
```
Comment (cclauss) [inline maths/prime_numbers.py]:
```suggestion
    >>> list(slow_primes(0))
    []
    >>> list(slow_primes(-1))
    []
    >>> list(slow_primes(-10))
    []
    >>> list(slow_primes(25))
    [2, 3, 5, 7, 11, 13, 17, 19, 23]
    >>> list(slow_primes(11))
    [2, 3, 5, 7, 11]
    >>> list(slow_primes(33))
    [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31]
    >>> list(slow_primes(10000))[-1]
```

Comment (steven7214) [reply maths/prime_numbers.py]: Sorry, I forgot to change these. It should be fixed now.

**Comment by TheSuperNoob** [comment]
These tests do not work as intended. The `primes` and `slow_primes` are generators and will need to be looped through fully in order to time them properly.

Testing the old compared to the new properly you can see that the performance gain is quite significant.

Screenshot 2020-05-15 at 00 57 56

**Comment by steven7214** [comment]
@TheSuperNoob How exactly are you testing them? Could you update the benchmark to better represent the performance gain?

**Pull request merged**